pkgs/io_file/lib/src/file_system.dart

```diff
@@ -30,6 +30,18 @@ class WriteMode {
 
 /// An abstract representation of a file system.
 base class FileSystem {
+  /// Checks whether two paths refer to the same object in the file system.
+  ///
+  /// Throws `PathNotFoundException` if either path doesn't exist.
+  ///
+  /// Links are resolved before determining if the paths refer to the same
+  /// object. Throws `PathNotFoundException` if either path requires resolving
+  /// a broken link.
+  /// ```
+  bool same(String path1, String path2) {
+    throw UnsupportedError('same');
+  }
+
```

Comment:
Maybe just not have this method, if it's not supported on all file systems. 
Only have it on `PosixFileSystem`.

If you have to do `is PosixFileSystem` before you can use it safely, better just have to do that before being able to use it at all.

If you plan to implement it on all file systems, make it abstract here, to force an implementation.

Reply:
It will be implemented on all file systems.